### PR TITLE
added modal to current state of our app

### PR DESCRIPTION
**Pull request by maraquepo**
@haydennharper 
I have the modal added with the dependencies for mui/styled-component. I also upgrade react/react-dom to the latest version since I was getting compatibility issues w/ mui

### CODE REVIEW

File: client/src/components/RightComponent/NYTModal/NYTModal.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
+
+let result = sample2.results
+
+let categories = result.map((cat) => {
+  return cat.list_name
+})
```
Comment (ks10825n) [inline client/src/components/RightComponent/NYTModal/NYTModal.jsx:27]: I think it is bad practice to create variables within the global scope like this. Will it break anything if you move this inside your component?

File: client/src/components/RightComponent/NYTModal/NYTModal.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+console.log(categories)
```
Comment (ks10825n) [inline client/src/components/RightComponent/NYTModal/NYTModal.jsx:29]: delete? Or make a reminder like TODO: to delete?

File: client/src/components/RightComponent/NYTModal/NYTModal.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  const {children, value, index} = props;
+
+  return(
+    <div>
+      {value === index && (
+        <h1>{children}</h1>
+      )}
+    </div>
+  )
+}
```
Comment (ks10825n) [inline client/src/components/RightComponent/NYTModal/NYTModal.jsx:86]: Can we break this out into another file and reimport it in here? If our component folder is getting too hefty from too many components, we can discuss file structure again as a group? Open to hearing the rest of the group's ideas on how to keep our clutter clean.

File: package.json [hunk 4]
```diff
@@ -21,19 +21,23 @@
...
     "bootstrap": "^5.1.3",
     "connect-mongodb-session": "^3.1.1",
     "express": "^4.17.2",
     "express-session": "^1.17.2",
     "mongoose": "^6.1.8",
-    "react": "^16.8.0",
+    "react": "^17.0.0",
     "react-bootstrap": "^2.1.1",
-    "react-dom": "^16.8.0",
+    "react-dom": "^17.0.0",
```
Comment (ks10825n) [inline package.json:36]: I don't think we have any peer dependencies that will break if we upgrade to v17. I think we went with v16.8 as it has more stable testing libraries for v16, but I don't think we'll get to any testing at all. Let's hear back from the rest of our group if there's any concerns in upgrading to v17. If there are you can always down grade material ui to this
[https://v4.mui.com/getting-started/installation/](url)

**Review by ks10825n — COMMENTED** [review]
Hey Matt, looks good. Left a few comments. Let's wait to get another review on this

File: client/src/components/RightComponent/NYTModal/NYTModal.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import { Modal, Button, Typography, Box, Tabs, Tab } from '@mui/material';
+import styled from 'styled-components';
+import sample2 from './sample.js'
+import ModalCards from './ModalCards.jsx';
+import { TabPanel } from './TabPanel.jsx';
+
+
+const NYTModal = () => {
+  const [show, setShow] = useState(false);
```
Comment (anerolabs) [inline client/src/components/RightComponent/NYTModal/NYTModal.jsx:10]: show and setShow feels a bit ambiguous- perhaps rename to showModal and setShowModal or similar for more clarity?

File: client/src/components/RightComponent/NYTModal/TabPanel.jsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+export const TabPanel = (props) => {
+  const {children, value, index} = props;
```
Comment (anerolabs) [inline client/src/components/RightComponent/NYTModal/TabPanel.jsx:4]: Tiny thing but we could lose line 4 if you destructure props as arguments right in line 3

File: client/src/components/RightComponent/RightComponent.jsx [hunk 7]
```diff
@@ -7,12 +7,13 @@ import SplitButton from 'react-bootstrap/SplitButton';
 import ButtonGroup from 'react-bootstrap/ButtonGroup';
```
Comment (anerolabs) [inline client/src/components/RightComponent/RightComponent.jsx:7]: Some imports here that I'm not seeing used in the component- are they for future revisions?

File: client/src/components/RightComponent/NYTModal/NYTModal.jsx (hunk 1 above)
Comment (brynrmrzz) [reply client/src/components/RightComponent/NYTModal/NYTModal.jsx]: I would say may be worth while to just move all styling to the css file to keep it consistent.

**Review by brynrmrzz — APPROVED** [review]

**Pull request closed without merging**